src/SignalR/clients/csharp/Client.Core/src/HubConnection.cs

```diff
@@ -676,7 +679,8 @@ private async Task<(bool close, Exception exception)> ProcessMessagesAsync(HubMe
                     break;
                 case InvocationMessage invocation:
                     Log.ReceivedInvocation(_logger, invocation.InvocationId, invocation.Target, invocation.Arguments);
-                    await DispatchInvocationAsync(invocation);
+                    // No need to wait on the On handlers logic. Would prevent us from awaiting other invocations.
+                    _ = taskQueue.Enqueue((state) => DispatchInvocationAsync((InvocationMessage)state), invocation);
```

Comment:
No closures....

Reply:
Going with a different approach now.  Going to Use a `Channel<Func<object, task>>` as the underlying queue and instead directly queue the tasks from `handler.InvokeAsync`  rather than the task from `DispatchInvocationAsync`

Reply:
Just use a Channel<InvocationMessage>

Reply:
Yeah, I think that simplifies this a lot. Let me update this

Reply:
The remaining question is whether we want On callbacks to wait for the Tasks of all previous On callbacks to complete. Obviously unblocking the ProcessMessagesAsync loop is more important, but it might be a good idea to unblock the client invocation loop as well.

Reply:
Lets leave it blocking for now. We'll only invoke a single callback at a time and that will be the contract.